src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -0,0 +1,323 @@
+package plugins;
+
+import jakarta.inject.Inject;
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.GroupApi;
+import org.gitlab4j.api.ProjectApi;
+import org.gitlab4j.api.models.*;
+import org.jenkinsci.test.acceptance.Matchers;
+import org.jenkinsci.test.acceptance.docker.DockerContainerHolder;
+import org.jenkinsci.test.acceptance.docker.fixtures.GitLabContainer;
+import org.jenkinsci.test.acceptance.junit.*;
+import org.jenkinsci.test.acceptance.plugins.credentials.CredentialsPage;
+import org.jenkinsci.test.acceptance.plugins.credentials.ManagedCredentials;
+import org.jenkinsci.test.acceptance.plugins.gitlab_plugin.*;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+import org.jenkinsci.test.acceptance.po.WorkflowMultiBranchJob;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.net.http.HttpResponse;
+
+import java.io.IOException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.jenkinsci.test.acceptance.Matchers.hasContent;
+import static org.junit.Assert.*;
+
+@WithDocker
+@Category(DockerTest.class)
+@WithPlugins({"gitlab-branch-source", "workflow-multibranch"})
+public class GitLabPluginTest extends AbstractJUnitTest {
+
+    @Inject
+    DockerContainerHolder<GitLabContainer> gitLabServer;
+
+    private GitLabContainer container;
+    private String host;
+    private int port;
+
+    private String privateTokenAdmin;
+
+    private String privateTokenUser;
+
+    private String repoName = "testrepo";
+
+    private String anotherRepoName = "anotherproject";
+
+    private String adminUserName = "testadmin";
+
+    private String userName = "testsimple";
+
+    private String groupName = "firstgroup";
+
+    private String JenkinsfileContent = "pipeline {\n" +
```

Comment:
This pipeline is using declarative syntax which should mean you need to declare the `pipeline-model-definition` plugin in the `@WithPlugins` annotation.